clang/include/clang/Driver/Options.td

```diff
@@ -3004,6 +3004,10 @@ def fsycl_max_parallel_jobs_EQ : Joined<["-"], "fsycl-max-parallel-link-jobs=">,
   "or AOT compilation of each device image.">;
 def : Flag<["-"], "fsycl-rdc">, Flags<[CoreOption]>, Alias<fgpu_rdc>;
 def : Flag<["-"], "fno-sycl-rdc">, Flags<[CoreOption]>, Alias<fno_gpu_rdc>;
+def fsycl_optimize_non_user_code : Flag<["-"], "fsycl-optimize-non-user-code">,
```

Comment:
A better name for this option would be 'fsycl-optimize-sycl-framework-functions'. if there has already been discussion on this, please feel free to ignore my suggestion. Thanks.